sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/management_link.py

```diff
@@ -157,6 +159,7 @@ def _on_message_received(self, _, message):
             to_remove_operation.on_execute_operation_complete(
                 mgmt_result, status_code, status_description, message, response_detail.get(b"error-condition")
             )
+        with self.lock:
```

Comment:
qq do we still get that list error even with the correlation id fix?

Reply:
Not that I've seen, but this seems like a good thing to lock still since, with multiple threads, the same op may try to be removed. Any concerns?

Reply:
nope!